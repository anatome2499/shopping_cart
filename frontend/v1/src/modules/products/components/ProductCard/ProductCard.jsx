import React from 'react'

import Grid from '@material-ui/core/Grid'

const ProductCard = ({
  imgUrl = 'https://www.mockuptiger.com/wp-content/uploads/2018/03/image-666.jpg',
  name,
  gender,
  age,
  price,
  availability
}) => {

  return (
    <Grid container>
      <Grid item sm={12}>
        <img src={imgUrl} />
      </Grid>

      <Grid item sm={8}>
        <p>Name: {name}</p>
        <p>Gender: {gender}</p>
        <p>Age: {age}</p>
      </Grid>

      <Grid item sm={4}>
        <h5>${price}</h5>
        <h5>{availability}</h5>
      </Grid>
    </Grid>
  )
}

export default ProductCard