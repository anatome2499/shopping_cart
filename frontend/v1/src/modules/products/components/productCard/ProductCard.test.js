import React from 'react'
import { shallow } from 'enzyme'

import ProductCard from './ProductCard'

describe('<ProductCard />', () => {
  let component

  beforeEach(() => {
    component = shallow(<ProductCard />)

  })

  it('should render with no errors', () => {
    
    expect(component).toMatchSnapshot()
  })

  it('should render with an image', () => {
    component.setProps({imgUrl: 'https://s3-ap-southeast-1.amazonaws.com/assets-eng-prod/products/th_310093-3.jpg'})

    expect(component).toMatchSnapshot()

    expect(component.find('img').length).toEqual(1)
  })

  it('should render with name, gender, age', () => {
    component.setProps({
      name: 'Princess Palace',
      gender: 'Female',
      age: '3_to_5'
    })

    expect(component).toMatchSnapshot()

    expect(component.find('p').length).toEqual(3)
  })

  it('should render with price and availability', () => {
    component.setProps({
      price: 24.95,
      availability: 'In Stock'
    })


    expect(component).toMatchSnapshot()

    expect(component.find('h5').length).toEqual(2)
  })
})