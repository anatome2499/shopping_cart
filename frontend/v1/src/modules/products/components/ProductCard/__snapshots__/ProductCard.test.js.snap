// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render with an image 1`] = `ShallowWrapper {}`;

exports[`<ProductCard /> should render with name, gender, age 1`] = `ShallowWrapper {}`;

exports[`<ProductCard /> should render with no errors 1`] = `ShallowWrapper {}`;

exports[`<ProductCard /> should render with price and availability 1`] = `ShallowWrapper {}`;
